org.eclipse.m2e.core/src/org/eclipse/m2e/core/project/ISettingsFiles.java

```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * Copyright (c) 2024 Christoph Läubrich and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Christoph Läubrich - initial API and implementation
+ ********************************************************************************/
+
+package org.eclipse.m2e.core.project;
+
+import java.io.File;
+
+
+/**
+ * Holds the locations of the maven global and user settings, global settings if specified are considered first but user
+ * settings might override.
+ */
+public interface ISettingsFiles {
+  /**
+   * Locates the global settings file, this takes <code>.mvn/maven.config</code> into account and otherwise returns the
+   * workspace configured one or even the maven default if no such settings are available.
```

Comment:
```suggestion
   * Locates the global settings file, this takes <code>.mvn/maven.config</code> into account and otherwise returns the
   * one configured in the workspace or even the maven default if no such settings are available.
```